client/state/ui/editor/reducer.js

```diff
@@ -21,8 +21,10 @@ import contactForm from './contact-form/reducer';
  */
 export function postId( state = null, action ) {
 	switch ( action.type ) {
-		case EDITOR_POST_ID_SET:
+		case EDITOR_START:
 			return action.postId;
+		case POST_SAVE_SUCCESS:
+			return state === action.postId ? action.savedPost.ID : state;
```

Comment:
Also, in theory, we could switch the editor's post before the SAVE_SUCCESS is dispatched. but very unlikely to happen.